airflow/utils/session.py

```diff
@@ -70,3 +70,29 @@ def wrapper(*args, **kwargs) -> RT:
                 return func(*args, session=session, **kwargs)
 
     return wrapper
+
+
+@provide_session
+@contextlib.contextmanager
+def create_global_lock(session=None, pg_lock_id=1, mysql_lock_name='init', mysql_lock_timeout=1800):
+    """Contextmanager that will create and teardown a global db lock."""
+    dialect = session.connection().dialect
+    if dialect.name == 'postgresql':
+        session.connection().execute(f'select PG_ADVISORY_LOCK({pg_lock_id});')
+
+    if dialect.name == 'mysql' and dialect.server_version_info >= (
+        5,
+        6,
+    ):
+        session.connection().execute(f"select GET_LOCK('{mysql_lock_name}',{mysql_lock_timeout});")
```

Comment:
One more comment. We have - as of recently - `mssql` support. In order to make it complete, we should support it too.

Reply:
Sure. Will need some time to prepare local setup for mssql

Reply:
With Breeze it's already there `--backend mssql` will bring the mssql database docker container up and configure Airflow to use use it .